test/hotspot/jtreg/serviceability/jvmti/vthread/PopFrameTest/PopFrameTest.java

```diff
@@ -144,7 +144,10 @@ public static void run() {
                 throwFailed("Main #B.2: expected JVMTI_ERROR_NONE");
             }
             log("Main #B.2: got expected JVMTI_ERROR_NONE");
+            notifyAtBreakpoint(); // needed to reset the native state
```

Comment:
This doesn't make much sense to me.  The purpose of `notifyAtBreakpoint()` is to unblock the Breakpoint event handler, which is waiting for this notify. However, the event handler can only ever be entered once, yet we have a previous `noitfyBreakpoint()` above and a 3rd `notifyAtBreakpoint()` below. It seems that this call and the one below are no-ops. I don't see how bp_sync_reached ever gets set true again after the first `notifyAtBreakpoint()` call is made.

Reply:
Thank you for the comment, Chris. You are right.
I've already figured my mistake/confusion (shared it on our team meeting) but tested my update which has been just pushed. My analysis in the bug report is incorrect, I'll update it soon. In fact, I don't really know what was the root cause of this deadlock because I can't reproduce the issue. Also, it is not easy to figure out by the code observation. The only idea I have is that it was a spurious wakeup on the `wait()` at line 52 in the Breakpoint handler of the native agent. I've added a couple of checks for robustness and added more tracing to get more details if this happen again.